### ISSUE TITLE
C# namespace separators are output in C++ style

### ISSUE THREAD

**Issue by tmakopan** [doxygen/doxygen#10511]
**Describe the bug**
Version1.9.6-1.9.8

Since the following change, C# argument types without links have been output in a C++-style namespace.

In the case of C#, it should be output with a C# style namespace separator.

```
SHA-1: ef95d88da5957bd96d0be3124b0d7a0f5086727c

* issue #9669 Namespaces in arguments of a method breaks cref resolving in C#

In case of CSharp convert the type from the dot notation to the (internally used) double colon notation.

```
![doxygen_issue](https://github.com/doxygen/doxygen/assets/11781325/da591ba8-6a57-4a0d-a408-beb3b91d3ad3)

**Comment by albert-github** [comment]
This looks like there is a missing conversion from the internal `::` to the `.` notation required for C#

- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).


I've been able to reproduce the problem, though an example is still useful to confirm my (not a C# programmers) assumption: [example.tar.gz](https://github.com/doxygen/doxygen/files/13770588/example.tar.gz)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10514

**Comment by tmakopan** [comment]
I confirmed that this issue has been resolved with patch #10514.
Thanks!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
